### PR TITLE
BLT-881: initial cut at developer skills documentation

### LINKED ISSUE #881: Improve Documentation to Address Recommended Developer Skillset
The goal of this task is to review the systems / knowledge required to interact with BLT on a minimum level, so that users understand what they are getting into (and have some resources to research).

### PR DESCRIPTION
**Pull request by mikemadison13**
Fixes #881 .

Changes proposed:
- updating Onboarding doc to link to Skills Doc
- Created Skills Doc that outlines minimum and suggested Skills for developers working with BLT

### CODE REVIEW

**Comment by mikemadison13** [comment]
@grasmash and @danepowell I would love eyes on this to ensure I haven't left out any resources that are worthwhile, misstated any objectives, etc.

**Comment by danepowell** [comment]
I like the idea of centralizing this kind of documentation, however I'm wondering if there's a less pointed way of linking to it from the onboarding doc. While it's true that developers _should_ eventually gain all of these skills, they aren't absolutely necessary if all you want to do is install a local Drupal site. I just don't want to make BLT seem unapproachable for newbies.

**Comment by mikemadison13** [comment]
@danepowell I agree that I don't want to scare people off, but I am trying to paint a realistic picture of "can you do these things?" if yes, then great. if no, then you should brush up or you'll have trouble.

File: readme/skills.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Minimum Skillset for Using BLT
+BLT is a tool that reaches far beyond Drupal development, and because of this developers planning on using it should working knowledge of the following technologies in order to be successful.
+
+See the [System Requirements](../INSTALL.md) for installing / using BLT. It is strongly urged that you use a Mac for local development, although certain versions of Linux and Windows 10 can also be used. Other operating systems ***should not*** be used for BLT development.
+ 
+## Enterprise Web Development
+### Virtual Environment
+#### [DrupalVM](https://www.drupalvm.com/)
+This project aims to make spinning up a simple local Drupal test/development environment incredibly quick and easy, and to introduce new developers to the wonderful world of Drupal development on local virtual machines (instead of crufty old MAMP/WAMP-based development).
```
Comment (grasmash) [inline readme/skills.md]: We don't require that people use VM with BLT. In fact, I use MAMP :)

I'd remove this or note that it isn't required.

File: readme/skills.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ * [Getting Started with Composer](https://getcomposer.org/doc/00-intro.md)
+ * [BLT Dependency Management](dependency-management.md)
+
+**Common Commands:**
+ * [composer install](https://getcomposer.org/doc/03-cli.md#install)
+ * [composer update](https://getcomposer.org/doc/03-cli.md#update)
+ * [composer require](https://getcomposer.org/doc/03-cli.md#require)
+ 
+####[NPM](https://www.npmjs.com/)
+npm is the package manager for JavaScript. Find, share, and reuse packages of code from hundreds of thousands of developers — and assemble them in powerful new ways.
```
Comment (grasmash) [inline readme/skills.md]: This is not part of BLT. It is an aspect of Cog. We need to indicate that this isn't required. Perhaps a separate "optional" section.

File: readme/skills.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * [Intro to PHP](http://php.net/manual/en/intro-whatis.php)
+ * [Using xdebug to Debug PHP](https://xdebug.org/docs/)
+ 
+ 
+We also strongly recommend an IDE for PHP Development Such as  
+ * [PHPStorm](https://www.jetbrains.com/phpstorm/)
+ * [Netbeans](http://netbeans.org/features/php/)
+
+##Front End Technologies
+###[Gulp](http://gulpjs.com/)
```
Comment (grasmash) [inline readme/skills.md]: Likewise this is Cog specific.

File: readme/skills.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  * [git checkout <branch>](https://git-scm.com/docs/git-checkout)
+  * [git commit](https://git-scm.com/docs/git-commit)
+  * [git commit --amend](https://git-scm.com/docs/git-commit#git-commit---amend)
+  * [git push](https://git-scm.com/docs/git-push)
+  * [git push -f](https://git-scm.com/docs/git-push#git-push--f)
+  * [git rebase](https://git-scm.com/docs/git-rebase)
+  * [git rebase -i](https://git-scm.com/docs/git-rebase#git-rebase---interactive)
+
+### Github
+GitHub is a web-based Git repository hosting service. It offers all of the distributed version control and source code management (SCM) functionality of Git as well as adding its own features. It provides access control and several collaboration features such as bug tracking, feature requests, task management, and wikis for every project.
```
Comment (grasmash) [inline readme/skills.md]: GitHub isn't required by BLT.

**Comment by mikemadison13** [comment]
@grasmash I have updated the language about a virtual environment (adding dev desktop, mentioning that it and drupal vm are optional). I also moved all of the NPM stuff / github down into the additional reading section (updated it to be optional) and added a note for all the front end stuff to reference cog and say it's not part of blt but is often bundled with it and its commands.

**Pull request merged**